### PR TITLE
Dal models and application context

### PR DESCRIPTION
**Pull request by AlKapsh**
Заметка касаемо контекста бд: явно указана строка подключения, она будет вынесена в файл appsettings.json после создания уровня представления.

### CODE REVIEW

File: Library.DAL/EFCore/ApplicationLibraryContext.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+        public ApplicationLibraryContext()
+        {
+            
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer("Server= .;Database=LibraryDB;Trusted_Connection=True;TrustServerCertificate=Yes;");
```
Comment (KatsiarynaAnashkevich) [inline Library.DAL/EFCore/ApplicationLibraryContext.cs:28]: Connection string should be in appsetings file

File: Library.DAL/EFCore/ApplicationLibraryContext.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+
```
Comment (KatsiarynaAnashkevich) [inline Library.DAL/EFCore/ApplicationLibraryContext.cs:33]: And one more moment. You should write entity configuration(fluent api)

File: Library.DAL/EFCore/ApplicationLibraryContext.cs [hunk 3]
```diff
@@ -17,15 +18,19 @@ internal class ApplicationLibraryContext: DbContext
...
         public ApplicationLibraryContext()
         {
             
         }
 
         protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
-            optionsBuilder.UseSqlServer("Server= .;Database=LibraryDB;Trusted_Connection=True;TrustServerCertificate=Yes;");
+            IConfigurationRoot configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
```
Comment (KatsiarynaAnashkevich) [inline Library.DAL/EFCore/ApplicationLibraryContext.cs:28]: It should be in extension method and after this in Program.cs

File: Library.DAL/Model Configuring/EntityConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            builder.Property(prop => prop.SecondName)
+                .IsRequired()
+                .HasMaxLength(30);
+
+            builder.HasMany(prop => prop.Books).WithMany(b => b.Authors);
+        }
+
+    }
+
+    internal class BookConfiguration : IEntityTypeConfiguration<Book>
```
Comment (KatsiarynaAnashkevich) [inline Library.DAL/Model Configuring/EntityConfiguration.cs:49]: All of this configurations should be in separate files

File: Library.DAL/EFCore/ApplicationLibraryContext.cs [hunk 5]
```diff
@@ -31,11 +30,17 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
 
             optionsBuilder.UseSqlServer(configuration.GetConnectionString("sqlConnection"));
+
+            
         }
 
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
-
+            modelBuilder.ApplyConfiguration(new AutorConfiguration());
```
Comment (KatsiarynaAnashkevich) [inline Library.DAL/EFCore/ApplicationLibraryContext.cs:39]: You can apply them just in one line of code

File: Library.DAL/EFCore/ApplicationLibraryContext.cs (hunk 3 above)
Comment (AlKapsh) [reply Library.DAL/EFCore/ApplicationLibraryContext.cs]: I know it. But now project don't have Program.cs, so i create extension method for options builder in last commit.

**Pull request closed without merging**